code/modules/bitrunning/objects/gimmick_disks/_gimmick_disk.dm

```diff
@@ -0,0 +1,112 @@
+
+/**
+ * Bitrunning tech disks which let you load full loadouts into the vdom on first avatar generation.
+ */
+/obj/item/bitrunning_disk/gimmick
+	desc = "A disk containing source code. It can be used to preload gimmick loadouts into the virtual domain."
+	/// The selected loadout that this grants
+	var/datum/bitrunning_gimmick/granted_loadout
+	/// The list of loadouts that this can grant
+	var/list/datum/bitrunning_gimmick/selectable_loadouts
+
+/obj/item/bitrunning_disk/gimmick/load_onto_avatar(mob/living/carbon/human/neo, mob/living/carbon/human/avatar, external_load_flags)
+	if(isnull(granted_loadout))
+		return LOAD_FAILED
+	return granted_loadout.grant_loadout(neo, avatar, external_load_flags)
+
+/obj/item/bitrunning_disk/gimmick/attack_self(mob/user, modifiers)
+	. = ..()
+
+	if(choice_made)
+		return
+
+	var/names = list()
+	for(var/datum/bitrunning_gimmick/loadout as anything in selectable_loadouts)
+		names += initial(loadout.name)
+
+	var/choice = tgui_input_list(user, message = "Select a gimmick loadout",  title = "Bitrunning Program", items = names)
+	if(isnull(choice) || !user.is_holding(src))
+		return
+
+	for(var/datum/bitrunning_gimmick/loadout as anything in selectable_loadouts)
+		if(initial(loadout.name) == choice)
+			granted_loadout = new loadout()
+
+	balloon_alert(user, "selected")
+	playsound(user, 'sound/items/click.ogg', 50, TRUE)
+	choice_made = choice
+
+
+/**
+ * The datum used by the gimmick loadout disk to determine what a loadout actually spawns.
+ */
+/datum/bitrunning_gimmick
+	/// Player readable name of the gimmick loadout
+	var/name = "Gimmick Loadout"
+	/// The list of actions that this will grant
+	var/list/datum/action/granted_actions
+	/// The list of items that this will grant
+	var/list/obj/item/granted_items
+	/// The item type we will use as a container for our granted items, given to the avatar
+	var/obj/item/container_item_type = /obj/item/storage/briefcase/secure/digital_storage
+	/// Prefix our name onto the
+	var/prefix_container_name = TRUE
+
+/// Grants out loadout.
+/datum/bitrunning_gimmick/proc/grant_loadout(mob/living/carbon/human/neo, mob/living/carbon/human/avatar, external_load_flags)
+	var/return_flags = NONE
+	return_flags |= grant_items(neo, avatar, external_load_flags)
+	return_flags |= grant_abilities(neo, avatar, external_load_flags)
+	return return_flags
```

Comment:
if one blocks load and the other doesn't does that mean we're in a partially loaded state? that's weird

Reply:
Yes! It _is_ weird, but it's intentional. Because a domain might block one, but not the other
It felt weirder to have one block type block the thing in its entirety, and we return an error flag anyway because _something_ failed to load and communicating that to the player is the primary goal here
Though in hindsight I feel I should update the warning messages to reflect partial loads as being possible

Reply:
do you intend on doing this?

Reply:
yes I simply did it, wasn't satisfied with the wording, and then forgot to do this